src/main/java/org/cyclonedx/model/Signature.java

```diff
@@ -0,0 +1,275 @@
+    ES256("ES256"),
+    @JsonProperty("ES384")
+    ES384("ES384"),
+    @JsonProperty("ES512")
+    ES512("ES512"),
+    @JsonProperty("Ed25519")
+    ED25519("Ed25519"),
+    @JsonProperty("Ed448")
+    ED448("Ed448"),
+    @JsonProperty("HS256")
+    HS256("HS256"),
+    @JsonProperty("HS384")
+    HS384("HS384"),
+    @JsonProperty("HS512")
+    HS512("HS512");
+
+    private final String name;
+
+    public String getAlgorithmName() {
+      return this.name;
+    }
+
+    Algorithm(String name) {
+      this.name = name;
+    }
+
+    public static Signature.Algorithm fromString(String text) {
+      for (Signature.Algorithm a : Signature.Algorithm.values()) {
+        if (a.name.equals(text)) {
+          return a;
+        }
+      }
+      return null;
+    }
+  }
+
+  @JsonAlias("chain")
+  private List<Signature> signers;
+  private Algorithm algorithm;
+  private String keyId;
+  private PublicKey publicKey;
+  private List<String> certificatePath;
+  private List<String> excludes;
+  private String value;
+
+  public List<Signature> getSigners() {
+    return signers;
+  }
+
+  public void setSigners(final List<Signature> signers) {
+    this.signers = signers;
+  }
+
+  public Algorithm getAlgorithm() {
+    return algorithm;
+  }
+
+  public void setAlgorithm(final Algorithm algorithm) {
+    this.algorithm = algorithm;
+  }
+
+  public String getKeyId() {
+    return keyId;
+  }
+
+  public void setKeyId(final String keyId) {
+    this.keyId = keyId;
+  }
+
+  public PublicKey getPublicKey() {
+    return publicKey;
+  }
+
+  public void setPublicKey(final PublicKey publicKey) {
+    this.publicKey = publicKey;
+  }
+
+  public List<String> getCertificatePath() {
+    return certificatePath;
+  }
+
+  public void setCertificatePath(final List<String> certificatePath) {
+    this.certificatePath = certificatePath;
+  }
+
+  public List<String> getExcludes() {
+    return excludes;
+  }
+
+  public void setExcludes(final List<String> excludes) {
+    this.excludes = excludes;
+  }
+
+  public String getValue() {
+    return value;
+  }
+
+  public void setValue(final String value) {
+    this.value = value;
+  }
+
+  public static class PublicKey {
+
+    public enum Kty {
+      @JsonProperty("EC")
+      EC("EC"),
+      @JsonProperty("OKP")
+      OKP("OKP"),
+      @JsonProperty("RSA")
+      RSA("RSA");
+
+      private final String name;
+
+      public String getKtyName() {
+        return this.name;
+      }
+
+      Kty(String name) {
+        this.name = name;
+      }
+
+      public static PublicKey.Kty fromString(String text) {
+        for (PublicKey.Kty k : PublicKey.Kty.values()) {
+          if (k.name.equals(text)) {
+            return k;
+          }
+        }
+        return null;
+      }
+    }
+
+    private Kty kty;
+
+    public enum Crv {
+      @JsonProperty("P-256")
+      P_256("P-256"),
+      @JsonProperty("P-384")
+      P_384("P-384"),
+      @JsonProperty("P-521")
+      P_521("P-521");
```

Comment:
I think there might be these other two as well
```suggestion
      P_521("P-521"),
      @JsonProperty("Ed25519")
      Ed25519("Ed25519"),
      @JsonProperty("Ed448")
      Ed448("Ed448");
```
https://cyclonedx.org/docs/1.4/json/#signature_oneOf_i2_publicKey_allOf_i1_then_crv